build.xml

```diff
@@ -59,14 +381,72 @@
 		</exec>
 	</target>
 
-	<!-- to option copy file to webapps :) -->
+	<target name="clear-tomcat" description="Delete Tomcat workdir">
+		<delete dir="${env.CATALINA_HOME}/work/Catalina/localhost"/>
+		<delete file="${env.CATALINA_HOME}/webapps/ScadaBR.war"/>
+		<delete dir="${env.CATALINA_HOME}/webapps/ScadaBR"/>
+	</target>
+
+	<target name="build" description="Perform project build workflow" depends="
+		init-core,
+		init-static-dependencies,
+		init-ui-dependencies,
+		build-core,
+		build-ui
+	"/>
+
+	<target name="build-no-ui" description="Build just a core application" depends="
+		init-core,
+		init-static-dependencies,
+		build-core"/>
+
+	<target name="deploy" description="Deploy application to Tomcat workdir" depends="
+		clear-tomcat,
+		deploy-war,
+		deploy-tomcat"/>
+
+	<target name="run-build" description="Build and run application" depends="
+		build,
+		deploy,
+		run
+	"/>
+
+	<target name="run-build-no-ui" description="Build and run application without building frontend application" depends="
+		build-no-ui,
+		deploy,
+		run
+	"/>
+
+	<target name="debug-build" depends="
```

Comment:
build-deploy-run-debug

Reply:
I use short names to make ease of use them instead of `ant build-deploy-run-debug` that is too long in my opinion. If you know the convention you will be able to understand that debug lunch the tomcat server and if you use keyword 'build' that means you must deploy it on tomcat so writing `deploy-run` is unnecessary.

Reply:
Okay, I'm writing this as an Intellij user and the length of the command doesn't matter that much, but I agree it might be redundant, and using with command. 
This may be a compromise: build-run-debug?

Reply:
But in our case we only "run" tomcat in debug mode using `catalina.sh jpda run` so you can see that is the same base command. I would add "run" keyword between "build" and "debug" if it would launch another script. But in this case it is only starting tomcat in debug mode. Intuitively you first look for the main task that you want to start. So using convention with "debug" on the beginning it is much simpler to keep the tasks organized.  We should create task names as short as it is possible. You can find an inspiration from [official Gradle doc](https://docs.gradle.org/current/userguide/command_line_interface.html). But keep in mind that there are just single words with tasks that depends on the previous ones. And here is the same convention for example you can create your own sequence of tasks to be done. It can be done using InteliiJ or just typing `ant build test create-war deploy` and it is very intuitive.

Imagine that situation that you want to add another "run" task that starts the JUnit tests before. It is much more intuitive if you create task called: "run-test" instead "test-run" because the first word is the one that is the most important and inform user what will be the final result. Tests are done during the whole process but the final result will be running application.